cmd/system-probe/api/client/client.go

```diff
@@ -35,3 +41,47 @@ func get(socketPath string) *http.Client {
 		},
 	}
 }
+
+// GetCheck returns data unmarshalled from JSON to T, from the specified module at the /<module>/check endpoint.
+func GetCheck[T any](client *http.Client, module types.ModuleName) (T, error) {
+	var data T
+	req, err := http.NewRequest("GET", ModuleURL(module, "/check"), nil)
+	if err != nil {
+		return data, err
+	}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return data, err
+	}
+	defer resp.Body.Close()
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return data, err
+	}
+	if resp.StatusCode != http.StatusOK {
+		return data, fmt.Errorf("non-ok status code: url %s, status_code: %d, response: `%s`", req.URL, resp.StatusCode, string(body))
+	}
+
+	err = json.Unmarshal(body, &data)
+	return data, err
+}
+
+func constructURL(module string, endpoint string) string {
+	u, _ := url.Parse("http://sysprobe")
```

Comment:
since the operation is constant and it won't change if `module` and `endpoint` will be different, there is no need to preform the redundant parsing, and instead let's create a simple object that contains whatever you need